libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/Journey.kt

```diff
@@ -44,6 +46,42 @@ class Journey private constructor(wp: Waypoints? = null) {
     fun plan(): String { return start().plan() }
     fun speed(): Int { return start().speed() }
     fun totalDistance(): Int { return end().totalDistance() }
+    fun totalTime(): Int { return end().totalTime() }
+
+    fun remainingDistance(distanceUntilTurn: Int): Int {
+        val actSeg = activeSegment()
+
+        if (actSeg != null) {
+            return if (activeSegment > 0) (totalDistance() - actSeg.cumulativeDistance + distanceUntilTurn)
+                // segments[0] is a summary of the whole journey and its Cumulative distance is same as Total distance
+            else (totalDistance()) // Don't try and calculate until actual segment has been found
+        }
+        else { return Int.MAX_VALUE }
+    }
+
+    fun remainingTime(distanceUntilTurn: Int): Int {
+        val actSeg = activeSegment()
+        val prevSeg = previousSegment()
+        val prevSegTime : Int
+        val timeToEndOfSeg : Int
+
+        if (prevSeg != null) {prevSegTime = prevSeg.time}
+        else {prevSegTime = 0}
+
+        if (actSeg != null) {
+            // Time to cover whole of active segment (Segment.time is cumulative time)
+            val segTime = actSeg.time - prevSegTime
+            if (actSeg.distance != 0) {
+                // This is an approximation of the time to the end of the active segment from current location
+                timeToEndOfSeg = ((segTime * distanceUntilTurn).toFloat() / actSeg.distance.toFloat()).toInt()
+            }
+            else {timeToEndOfSeg = 0}
```

Comment:
In this case I think
```kotlin
else {
    timetoEndOfSeg = 0
}
```
would be appropriate for readability, consistent with the style in the `if` (which in turn I think is correct)